src/hotspot/os/linux/os_linux.cpp

```diff
@@ -216,18 +216,27 @@ julong os::Linux::available_memory() {
 }
 
 julong os::physical_memory() {
-  jlong phys_mem = 0;
+  jlong phys_mem = Linux::physical_memory();
+  log_debug(os, container)("total physical memory: " JLONG_FORMAT, phys_mem);
+
   if (OSContainer::is_containerized()) {
-    jlong mem_limit;
-    if ((mem_limit = OSContainer::memory_limit_in_bytes()) > 0) {
+    jlong mem_limit = OSContainer::memory_limit_in_bytes();
+    if (mem_limit > 0 && mem_limit < phys_mem) {
       log_trace(os)("total container memory: " JLONG_FORMAT, mem_limit);
       return mem_limit;
     }
-    log_debug(os, container)("container memory limit %s: " JLONG_FORMAT ", using host value",
-                            mem_limit == OSCONTAINER_ERROR ? "failed" : "unlimited", mem_limit);
+    if (mem_limit >= phys_mem) {
+      log_debug(os, container)("container memory limit ignored: " JLONG_FORMAT ", using host value "
+                               JLONG_FORMAT, mem_limit, phys_mem);
+    } else if (OSCONTAINER_ERROR == mem_limit) {
+      log_debug(os, container)("container memory limit failed: " JLONG_FORMAT ", using host value "
+                               JLONG_FORMAT, mem_limit, phys_mem);
+    } else {
+      log_debug(os, container)("container memory limit unlimited: " JLONG_FORMAT ", using host value "
+                               JLONG_FORMAT, mem_limit, phys_mem);
+    }
```

Comment:
Can we consolidate these to:

```
const char* reason;
if (mem_limit >= phys_mem) {
    reason = "ignored";
} else ...
}

log_debug(os, container)("container memory limit %s: " JLONG_FORMAT ", using host value "
                               JLONG_FORMAT, reason, mem_limit, phys_mem);
```

Reply:
Certainly, thanks for the suggestion. Resolved in `7e64194b6b1ee04edfe5b51725e12f121094badc`.